core/schema/modulesource.go

```diff
@@ -145,25 +163,130 @@ func (s *moduleSchema) moduleSource(ctx context.Context, query *core.Query, args
 	return src, nil
 }
 
+// shared logic with originToPath, copy-pasted
+// to avoid dependency on Buildkit
+func resolveGitURL(ref string) (*gitutil.GitURL, error) {
+	u, err := gitutil.ParseURL(ref)
+	if err != nil {
+		if err == gitutil.ErrUnknownProtocol {
+			u, err = gitutil.ParseURL("https://" + ref)
+		}
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse URL: %w", err)
+		}
+	}
+	return u, nil
+}
+
 type parsedRefString struct {
 	modPath    string
 	modVersion string
 	hasVersion bool
-	isGitHub   bool
 	kind       core.ModuleSourceKind
+	repoRoot   *vcs.RepoRoot
 }
 
-func parseRefString(refString string) parsedRefString {
+// interface used for host interaction mocking
+type BuildkitClient interface {
```

Comment:
is it possible to make this interface private? specially because we probably don't want to expose it to other packages.

Reply:
Done 🙏

Reply:
Still seems to be public.

Reply:
yep, still public :D

Reply:
my bad, shall be done this time 😇